api/feedbooks.py

```diff
@@ -0,0 +1,457 @@
+import datetime
+import feedparser
+from nose.tools import set_trace
+from StringIO import StringIO
+from zipfile import ZipFile
+from lxml import etree
+import os
+from flask_babel import lazy_gettext as _
+
+from core.opds import OPDSFeed
+from core.opds_import import (
+    OPDSImporter,
+    OPDSImportMonitor,
+    OPDSXMLParser,
+)
+from core.model import (
+    DataSource,
+    ExternalIntegration,
+    Hyperlink,
+    Resource,
+    Representation,
+    RightsStatus,
+)
+from core.util.epub import EpubAccessor
+
+
+class FeedbooksOPDSImporter(OPDSImporter):
+
+    REALLY_IMPORT_KEY = 'really_import'
+    LANGUAGE_KEY = 'language'
+    REPLACEMENT_CSS_KEY = 'replacement_css'
+
+    NAME = ExternalIntegration.FEEDBOOKS
+    DESCRIPTION = _("Import open-access books from FeedBooks.")
+    SETTINGS = [
+        {
+            "key": REALLY_IMPORT_KEY,
+            "type": "select",
+            "label": _("Really?"),
+            "description": _("Most libraries are better off importing free Feedbooks titles via an OPDS Import integration from NYPL's open-access content server or DPLA's Open Bookshelf. This setting makes sure you didn't create this collection by accident and really want to import directly from Feedbooks."),
+            "options": [
+                { "key": "false", "label": _("Don't actually import directly from Feedbooks.") },
+                { "key": "true", "label": _("I know what I'm doing; import directly from Feedbooks.") },
+          ],
+          "default": "false"
+        },
+        {
+            "key": LANGUAGE_KEY,
+            "label": _("Import books in this language"),
+            "description": _("Feedbooks offers separate feeds for different languages. Each one can be made into a separate collection."),
+            "type": "select",
+            "options": [
+                { "key": "en", "label": _("English") },
+                { "key": "es", "label": _("Spanish") },
+                { "key": "fr", "label": _("French") },
+                { "key": "it", "label": _("Italian") },
+                { "key": "de", "label": _("German") },
+            ],
+            "default": "en",
+        },
+        {
+            "key" : REPLACEMENT_CSS_KEY,
+            "label": _("Replacement stylesheet"),
+            "description": _("This will replace the Feedbooks stylesheet with an alternate stylesheet in mirrored copies of the Feedbooks titles. The default value is an accessibility-focused stylesheet produced by the DAISY consortium."),
+            "default": "http://www.daisy.org/z3986/2005/dtbook.2005.basic.css",
+        },
```

Comment:
This was hard-coded in content.

Reply:
Will leaving it blank turn off CSS replacement?

Reply:
It should, yeah. I'll write a test for that.

Reply:
I'd also mention it in the description.